tools/rpkg/src/relational.cpp

```diff
@@ -280,12 +317,18 @@ static SEXP result_to_df(unique_ptr<QueryResult> res) {
 
 [[cpp11::register]] SEXP rapi_rel_set_intersect(duckdb::rel_extptr_t rel_a, duckdb::rel_extptr_t rel_b) {
 	auto res = std::make_shared<SetOpRelation>(rel_a->rel, rel_b->rel, SetOperationType::INTERSECT);
-	return make_external<RelationWrapper>("duckdb_relation", res);
+
+	cpp11::writable::list prot = {rel_a, rel_b};
+
+	return make_external_prot<RelationWrapper>("duckdb_relation", prot, res);
 }
 
 [[cpp11::register]] SEXP rapi_rel_set_diff(duckdb::rel_extptr_t rel_a, duckdb::rel_extptr_t rel_b) {
 	auto res = std::make_shared<SetOpRelation>(rel_a->rel, rel_b->rel, SetOperationType::EXCEPT);
-	return make_external<RelationWrapper>("duckdb_relation", res);
+
+	cpp11::writable::list prot = {rel_a, rel_b};
```

Comment:
Looks good, but can't we do the list wrapping in make_external_prot, too?

Reply:
Does this mean passing two arguments to `make_external_prot()`, or just inlining the `prot` variable?

Reply:
I was thinking of the latter but not sure if it still shows the desired behaviour

Reply:
I suspect it would, the temporary `list` won't be destroyed for the duration of the `make_external()`call, and hece will remain protected. We can try in a second iteration.